src/memory.rs

```diff
@@ -0,0 +1,63 @@
+use std::cell::Cell;
+use rand::rngs::StdRng;
+
+use rustc_mir::interpret::{Pointer, Allocation, AllocationExtra, InterpResult};
+use rustc_target::abi::Size;
+
+use crate::{stacked_borrows, intptrcast};
+use crate::stacked_borrows::{Tag, AccessKind};
+
+#[derive(Default, Clone, Debug)]
+pub struct MemoryState {
```

Comment:
Don't we call this `MemoryExtra` in librustc_mir? If so, please also call it that way here.

Reply:
I took the name from `stacked_borrows.rs`

Reply:
Well seems I picked poorly then. ;)